api/shape/gcode.js

```diff
@@ -0,0 +1,38 @@
+import { computeHash, emit, getSourceLocation, write } from '@jsxcad/sys';
+import { gridView, qualifyViewId } from './view.js';
+
+import { Shape } from './Shape.js';
+import { destructure } from './destructure.js';
+import { hash as hashGeometry } from '@jsxcad/geometry';
+import { toGcode } from '@jsxcad/convert-gcode';
+
+export const gcode = Shape.registerMethod(
+  'gcode',
+  (...args) =>
+    async (shape) => {
+      const {
+        value: name,
+        func: op = (s) => s,
+        object: options = {},
+      } = destructure(args);
+      const { id, path, viewId } = qualifyViewId(name, getSourceLocation());
+      let index = 0;
+      // Gcode doesn't do layout (although maybe it should)
```

Comment:
😛

Reply:
Ah, we can include layout now that it's segment based.